esp-hal/build.rs

```diff
@@ -11,16 +11,14 @@ use esp_build::assert_unique_used_features;
 use esp_config::{generate_config, Value};
 use esp_metadata::{Chip, Config};
 
-#[cfg(debug_assertions)]
```

Comment:
using `debug_assertions` to detect a debug build worked fine in the past but we care about the optimization level not `debug_assertions`